File: src/com/joeyang/Problem35.java
package com.joeyang;

public class Problem35 {
    public int searchInsert(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;

        while (left < right) {
            int mid = (left + right) >>> 1;
            if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        if (target <= nums[right]) {
            return right;
        } else {
            return right + 1;
        }
    }
}

File: src/com/joeyang/Problem35.kt
package com.joeyang

class Problem35 {
    fun searchInsert(nums: IntArray, target: Int): Int {
        if (nums.isEmpty()) {
            return 0
        }

        var left = 0
        var right = nums.size
        while (left < right) {
            val mid = (left + right) / 2
//            if (nums[mid] < target) {
//                left = mid + 1
//            } else {
//                right = mid
//            }
            if (target < nums[mid]) {
                right = mid
            } else {
                left = mid
            }
        }
        return left
    }
}